src/Stratis.Features.SQLiteWalletRepository/Tables/HDTransactionData.cs

```diff
@@ -245,37 +245,47 @@ internal static IEnumerable<FlattenedHistoryItem> GetHistory(DBConnection conn,
                     -- Find all receives
                     SELECT  t.OutputTxId as Id
                     ,       t.RedeemScript
-                    ,       CASE    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN 0
-                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN 3
-                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN 2
+                    ,       CASE    WHEN t.OutputTxIsCoinbase = 0 THEN 0                                       -- Received
+                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN 3                 -- Mined
+                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN 2                -- Staked
                             END Type                 
                     ,       t.OutputTxTime as TimeStamp
-                    ,       CASE    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN t.Value
-                                    WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 1 THEN ((SELECT sum(tt.Value) FROM HDTransactionData tt WHERE tt.SpendTxId = t.OutputTxId) - t.Value)
-                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN t.Value
-                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN (SUM(t.Value) - (                                                                
+                    ,       CASE    WHEN t.OutputTxIsCoinbase = 0 THEN SUM(t.Value)                             -- Received
+                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN SUM(t.Value)       -- Mined
+                                    WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN SUM(t.Value) - IFNULL(( -- Staked
                                         SELECT ttp.Value
                                         FROM HDPayment p
                                         INNER JOIN HDTransactionData ttp ON ttp.OutputTxId = p.OutputTxId AND ttp.OutputIndex = p.OutputIndex AND ttp.WalletId = {strWalletId} AND ttp.AccountIndex = {strAccountIndex}
                                         WHERE p.SpendTxId = t.OutputTxId AND p.SpendIsChange = 0
                                         LIMIT 1
-                                    ))
+                                    ), 0)
                             END Amount
                     ,       NULL as Fee
                     ,       NULL as SendToScriptPubkey
                     ,       t.Address AS ReceiveAddress
                     ,       t.OutputBlockHeight as BlockHeight
                     FROM    HDTransactionData AS t
                     WHERE   t.WalletId = {strWalletId} AND t.AccountIndex = {strAccountIndex}{((address == null) ? "" : $@" AND t.Address = {strAddress}")}
-                    GROUP   BY t.OutputTxId
+                    AND     t.AddressType = 0
```

Comment:
I did a compare between the output of `release/1.0.9.1` and your branch with my local STRAX wallet and noticed that some staking transactions were missing:

![image](https://user-images.githubusercontent.com/4281271/128193146-7be741c0-e99c-4846-abbc-41a03a4ecba9.png)

It appears that with that particular tx the address type was 1 (change) and as such they are getting filtered out if you include the above. Perhaps that needs to change to take the coinbase into account?